linux:drivers: phy, vhub and usb driver update

ashok2ku opened the pull request:
Updated latest driver from AspeedTech branch.
phy,usb and vhub for keyboard and mouse support

Tested: on Congo with A1

aasboddu committed:
ARM64:dts:aspeed:Congo: Add BIOS/FPGA flash memory

Add SPI2 controller for BIOS/FPGA flash memory
access. Updated clock speed to 33MHz

Signed-off-by: aasboddu <Aashray.Boddu@amd.com>

aasboddu committed:
ARM64:dts:aspeed:Congo: Modify I2C Bus for Congo A1

Modify I2C Bus number for Congo based on
A1 silicon specification.

Signed-off-by: aasboddu <Aashray.Boddu@amd.com>

aasboddu committed:
ARM64:dts:aspeed:Congo: Add USB vHUB for KVM

Add USB vHUB to enable keyboard/mouse for KVM

Signed-off-by: aasboddu <Aashray.Boddu@amd.com>

aasboddu committed:
ARM64:dts:aspeed:Morocco: Fix memory map

Fix node name in memory mapping for A1 Silicon

Signed-off-by: aasboddu <Aashray.Boddu@amd.com>

aasboddu committed:
ARM64:dts:aspeed:Morocco: add vBIOS/xDMA for KVM

Add vBIOS and xDMA memory to enable KVM for A1
silicon.

Signed-off-by: aasboddu <Aashray.Boddu@amd.com>

aasboddu committed:
ARM64:dts:aspeed:Morocco: Add BIOS/FPGA flash memory

Add SPI2 controller for BIOS/FPGA flash memory
access. Updated clock speed for SPI devices to 33MHz.

Signed-off-by: aasboddu <Aashray.Boddu@amd.com>

aasboddu committed:
ARM64:dts:aspeed:Morocco: Modify I2C Bus for Morocco A1

Modify I2C Bus number for Morocco based on
A1 silicon specification.

Signed-off-by: aasboddu <Aashray.Boddu@amd.com>

aasboddu committed:
ARM64:dts:aspeed:Morocco: Add A1 LTPI GPIO mapping

Add LTPI GPIO mapping based A1 Silicon.

Signed-off-by: aasboddu <Aashray.Boddu@amd.com>

aasboddu committed:
ARM64:dts:aspeed:Morocco: Add A1 LTPI UART mapping

Add LTPI UART mapping based A1 Silicon.

Signed-off-by: aasboddu <Aashray.Boddu@amd.com>

aasboddu committed:
ARM64:dts:aspeed:Morocco: Add USB vHUB for KVM

Add USB vHUB to enable keyboard/mouse for KVM

Signed-off-by: aasboddu <Aashray.Boddu@amd.com>

aasboddu committed:
ARM64:dts:aspeed:Morocco: Add SGPIO for A1 silicon

Add SGPIO support for A1 silicon.

Signed-off-by: aasboddu <Aashray.Boddu@amd.com>

ashok2ku committed:
linux:drivers: phy, vhub and usb driver update

Updated latest driver from AspeedTech branch.
phy,usb and vhub for keyboard and mouse support

Tested: on Congo with A1

Signed-off-by: ashok2ku <Ashok2.Kumar@amd.com>

aasboddu committed:
ARM64:dts:aspeed:Kenya: Fix memory map

Fix node name in memory mapping for A1 Silicon

Signed-off-by: aasboddu <Aashray.Boddu@amd.com>

aasboddu committed:
ARM64:dts:aspeed:Kenya: add vBIOS/xDMA for KVM

Add vBIOS and xDMA memory to enable KVM for A1
silicon.

Signed-off-by: aasboddu <Aashray.Boddu@amd.com>

aasboddu committed:
ARM64:dts:aspeed:Kenya: Add BIOS/FPGA flash memory

Add SPI2 controller for BIOS/FPGA flash memory
access. Updated clock speed for SPI devices to 33MHz.

Signed-off-by: aasboddu <Aashray.Boddu@amd.com>

aasboddu committed:
ARM64:dts:aspeed:Kenya: Modify I2C Bus for Kenya A1

Modify I2C Bus number for Kenya based on
A1 silicon specification.

Signed-off-by: aasboddu <Aashray.Boddu@amd.com>

aasboddu committed:
ARM64:dts:aspeed:Kenya: Add A1 LTPI GPIO mapping

Add LTPI GPIO mapping based A1 Silicon.

Signed-off-by: aasboddu <Aashray.Boddu@amd.com>

aasboddu committed:
ARM64:dts:aspeed:Kenya: Add SGPIO for A1 silicon

Add SGPIO support for A1 silicon.

Signed-off-by: aasboddu <Aashray.Boddu@amd.com>

aasboddu committed:
ARM64:dts:aspeed:Morocco: Add GPIO HOG for eSPI

Add GPIO HOG to set eSPI GPIO line to low and
direction to output enabling BMC to receive BIOS
POST codes for A1 Silicon.

Signed-off-by: aasboddu <Aashray.Boddu@amd.com>

aasboddu committed:
ARM64:dts:aspeed:Kenya: Add GPIO HOG for eSPI

Add GPIO HOG to set eSPI GPIO line to low and
direction to output enabling BMC to receive BIOS
POST codes for A1 Silicon.

Signed-off-by: aasboddu <Aashray.Boddu@amd.com>

aasboddu committed:
ARM64:dts:aspeed:Nigeria: Fix memory map

Fix node name in memory mapping for A1 Silicon

Signed-off-by: aasboddu <Aashray.Boddu@amd.com>

aasboddu committed:
ARM64:dts:aspeed:Nigeria: add vBIOS/xDMA for KVM

Add vBIOS and xDMA memory to enable KVM for A1
silicon.

Signed-off-by: aasboddu <Aashray.Boddu@amd.com>

aasboddu committed:
ARM64:dts:aspeed:Nigeria: Add BIOS/FPGA flash memory

Add SPI2 controller for BIOS/FPGA flash memory
access. Updated clock speed for SPI devices to 33MHz.

Signed-off-by: aasboddu <Aashray.Boddu@amd.com>

aasboddu committed:
ARM64:dts:aspeed:Nigeria: Modify I2C Bus for A1

Modify I2C Bus number for Nigeria based on
A1 silicon specification.

Signed-off-by: aasboddu <Aashray.Boddu@amd.com>

aasboddu committed:
ARM64:dts:aspeed:Nigeria: Add A1 LTPI GPIO mapping

Add LTPI UART mapping based A1 Silicon.

Signed-off-by: aasboddu <Aashray.Boddu@amd.com>

aasboddu committed:
ARM64:dts:aspeed:Nigeria: Add A1 LTPI UART mapping

Add LTPI UART mapping based A1 Silicon.

Signed-off-by: aasboddu <Aashray.Boddu@amd.com>

aasboddu committed:
ARM64:dts:aspeed:Nigeria: Add USB vHUB for KVM

Add USB vHUB to enable keyboard/mouse for KVM

Signed-off-by: aasboddu <Aashray.Boddu@amd.com>

aasboddu committed:
ARM64:dts:aspeed:Nigeria: Add SGPIO for A1 silicon

Add SGPIO support for A1 silicon.

Signed-off-by: aasboddu <Aashray.Boddu@amd.com>

aasboddu committed:
ARM64:dts:aspeed:Nigeria: Add GPIO HOG for eSPI

Add GPIO HOG to set eSPI GPIO line to low and
direction to output enabling BMC to receive BIOS
POST codes for A1 Silicon.

Signed-off-by: aasboddu <Aashray.Boddu@amd.com>

ojayanth commented:
It will be good to Keep separate commit for each driver. Like driver specific commit includes
  - Cherry pick initial version
  - Update driver source code with latest version and append commit message with the  commit reference. commit id or aspeed release version information.

aasboddu committed:
ARM64:dts:aspeed:Congo: Add GPIO HOG for eSPI

Add GPIO HOG to set eSPI GPIO line to low and
direction to output enabling BMC to receive BIOS
POST codes for A1 Silicon.

Signed-off-by: aasboddu <Aashray.Boddu@amd.com>

aasboddu committed:
ARM64:dts:aspeed:Kenya: Add A1 LTPI UART mapping

Add LTPI UART mapping based A1 Silicon.

Signed-off-by: aasboddu <Aashray.Boddu@amd.com>

aasboddu committed:
ARM64:dts:aspeed:Kenya: Add USB vHUB for KVM

Add USB vHUB to enable keyboard/mouse for KVM

Signed-off-by: aasboddu <Aashray.Boddu@amd.com>

ashok2ku committed:
Update aspeed-bmc-amd-congo.dts

As per Raja/Sreedhar - we dont need this changes for now

modolaty committed:
ARM64:dts:aspeed A1 DTS changes for BU

Add "P0_MGMT_ASSERT_WARM_RST_BTN_L" as LTPI GPIO
Add SBRMI and SBTSI new devices for i3c4 and i3c5

Tested: verified in Morocco with A1 BMC and Verona

Signed-off-by: modolaty <mohsen.dolaty@amd.com>

modolaty committed:
linux:drivers: i3c and amd-apml driver for SP7 BU

Add changes to handle the new device IDs for SP7
change i3c device.c
change amd-apml SBTSI and SBRMI driver
Add debug messages to mipi CCC cmd processing

Tested: verified in Congo A0 IOD, and Morocco A1 with Verona

Signed-off-by: modolaty <mohsen.dolaty@amd.com>

ashok2ku commented:
placed new pull request 
https://github.com/AMDESE/linux-aspeed/pull/83
